wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -2405,17 +2415,36 @@ instance BuildableSafeGen SlotDuration where
 data NodeSettings = NodeSettings {
      setSlotDuration   :: !SlotDuration
    , setSoftwareInfo   :: !(V1 Core.SoftwareVersion)
-   , setProjectVersion :: !Version
+   , setProjectVersion :: !(V1 Version)
    , setGitRevision    :: !Text
    } deriving (Show, Eq, Generic)
 
 #if !(MIN_VERSION_swagger2(2,2,2))
 -- See note [Version Orphan]
-instance ToSchema Version where
+instance ToSchema (V1 Version) where
     declareNamedSchema _ =
-        pure $ NamedSchema (Just "Version") $ mempty
+        pure $ NamedSchema (Just "V1Version") $ mempty
             & type_ .~ SwaggerString
 
+instance Buildable (V1 Version) where
+    build (V1 v) = bprint shown v
+
+instance Buildable (SecureLog (V1 Version)) where
+    build (SecureLog x) = Formatting.Buildable.build x
+
+instance ToJSON (V1 Version) where
+    toJSON (V1 v) = toJSON (showVersion v)
+
+instance FromJSON (V1 Version) where
+    parseJSON (String v) = case readP_to_S parseVersion (T.unpack v) of
+        (reverse -> ((ver,_):_)) -> pure (V1 ver)
+        _                        -> mempty
+    parseJSON x                  = typeMismatch "Not a valid Version" x
+
+instance Arbitrary (V1 Version) where
+    arbitrary = fmap V1 arbitrary
```

Comment:
Where does the `Arbitrary Version` comes from ?

Reply:
http://hackage.haskell.org/package/QuickCheck-2.12.6.1/docs/src/Test.QuickCheck.Arbitrary.html#line-927

Reply:
Ho :o ! 
That's a surprise. Great :tada: